hooks/read-package-hook/src/createOptionalDependenciesRemover.ts

```diff
@@ -0,0 +1,15 @@
+import type { BaseManifest, ReadPackageHook } from '@pnpm/types'
+
+export function createOptionalDependenciesRemover (toBeRemoved: string[]): ReadPackageHook {
+  return <Manifest extends BaseManifest> (manifest: Manifest) => removeOptionalDependencies(manifest, toBeRemoved)
+}
+
+function removeOptionalDependencies<Manifest extends BaseManifest> (manifest: Manifest, toBeRemoved: string[]): Manifest {
+  for (const optionalDependency in manifest.optionalDependencies) {
+    if (toBeRemoved.includes(optionalDependency)) {
```

Comment:
probably we need to support pattern matching. Same way like here: https://github.com/pnpm/pnpm/blob/0e6b757cf58a324ab9bffd3e573a2982f199e806/hooks/read-package-hook/src/createPeerDependencyPatcher.ts#L13